metrics/web/docs/features/storybook/01_metrics_storybook_design.md

```diff
@@ -0,0 +1,199 @@
+ - *base* - a widget that is responsible for only displaying the given data. These widgets should be highly configurable and usable out of the Metrics Web Application context.
+ - *common* - is a widget that is actually used in the Metrics Web Application context and can be used across the modules.
+
+The main question is whether we want to extract to the Metrics storybook only base widgets or add common widgets as well. If we add only base widgets to the storybook, we can accidentally duplicate the existing common Metrics widgets with the same functionality. Also, we will not be able to view the entire list of possible components that are used in the application. That's why our choice is to take out base widgets together with common ones.
+
+At the same time, some of the common Metrics widgets (e.g., [MetricsUserMenu](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/lib/common/presentation/widgets/metrics_user_menu.dart)) are using [ChangeNotifiers](https://pub.dev/documentation/provider/latest/provider/ChangeNotifierProvider-class.html) internally. It makes widgets dependent on the Metrics Web Application's notifiers, so in these cases, we cannot extract this kind of widget, without changing them.
+
+According to the [widget structure organization document](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#applying-a-theme-to-a-widget-appearance) common widgets use the Metrics theme. Thus to display these widgets in the storybook correctly we should move the Metrics theme along with them. The disadvantage of moving common widgets is the loss of the ability to customize them as base widgets.
+
+Also, as the common Metrics widgets use the Metrics theme, the storybook should have the ability to toggle between the light and dark variants of the theme, to affect widgets' appearance.
+
+Now, we can list the requirements the `Metrics storybook` feature must satisfy:
+
+- Displays a set of Metrics widgets.
+- Provides an ability to manage the Metrics theme.
+- Contains a tab with the widget documentation.
+- Contains a list of inputs to change widgets' appearance (height, width, color, etc.).
+- Contains the Metrics color palette to view all project's colors.
+- Provides an ability to search across components.
+- Visually groups widgets by their types.
+- Provides an ability to change a viewport size (view widgets on different devices).
+
+### Landscape
+> Look for existing solutions in the area.
+
+There are two approaches to create the storybook for Flutter widgets - [using existing packages](#use-existing-packages) or [creating a UI from scratch](#manually-create-a-storybook-from-scratch).
+
+#### Use existing packages
+
+At this time, there are only a few packages, that provide the functionality to write Storybook for Flutter widgets.
+
+Let's take a closer look at the packages:
+
+1. [Storyboard](https://pub.dev/packages/storyboard)
```

Comment:
:readable: Would it be great to convert this numbered list into separate sub-sections?